### PR TITLE
Fix tcpip constraints

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by avsm** [comment]
i think a lot of the older versions need `io-page < 2.0.0` or `io-page-xen` or `io-page-unix` added...

File: packages/arp/arp.0.1.1/opam [hunk 1]
```diff
@@ -29,6 +29,11 @@ depopts: [
   "tcpip"
   "lwt"
 ]
+
+conflicts: [
+  "tcpip" {< "3.0.0"}
+]
```
Comment (hannesm) [inline packages/arp/arp.0.1.1/opam]: that's wrong.. `arp < 0.2.0` should conflict with `tcpip  > 3.0.0`, not `<`... only `arp=0.2.0` got support for `tcpip-3.0.0`

Comment (samoht) [inline packages/arp/arp.0.1.1/opam]:
```
$ opam install arp.0.1.1 'tcpip<3.0.0'
...
∗  installed tcpip.2.8.1
[ERROR] The compilation of arp failed at "ocaml pkg/pkg.ml build --pinned false
        --with-mirage true".
...
# + ocamlfind ocamlopt -c -g -annot -bin-annot -safe-string -principal -package cstruct -package ipaddr -package logs -package mirage-types -package lwt -package tcpip.ethif -package result -w +A-4-6-44-48 -warn-error +1..49 -I mirage -I src -o mirage/marp.cmx mirage/marp.ml
# File "mirage/marp.ml", line 43, characters 23-35:
# Error: Unbound module Ethif_packet

$ opam install arp.0.1.1 'tcpip>=3.0.0'
...
∗  installed tcpip.3.1.3
∗  installed arp.0.1.1
Done.
```

**Comment by samoht** [comment]
Revdeps are now passing. Merging quickly to avoid hammering the CI infrastructure.

File: packages/arp/arp.0.1.1/opam (hunk 1 above)
Comment (hannesm) [inline packages/arp/arp.0.1.1/opam]: oh, ic - thanks for investigating (I remember I wrote arp with the then unreleased move of tcpip into smaller libraries in mind).  my intuition of `tcpip >= 3` and `mirage-types-lwt <3` is that this should be empty (post-mirage3 and pre-mirage3 things)... maybe best to remove `arp < 0.2.0`!?

Comment (samoht) [inline packages/arp/arp.0.1.1/opam]: the intersection is not empty apparently :-)

Comment (hannesm) [inline packages/arp/arp.0.1.1/opam]: is this a bug or a feature?

Comment (samoht) [inline packages/arp/arp.0.1.1/opam]: It's ok I think, the old `mirage-type` can be installed with newer `mirage-protocols` package. Seems a good feature to me!

**Pull request merged**